app/models/piece.rb

```diff
@@ -2,75 +2,62 @@
 
 class Piece < ApplicationRecord
   belongs_to :game
+  
+  def is_white?
+    piece_number < 6
+  end
+
+  def is_black?
+    piece_number > 5
+  end
 
   def is_obstructed?(x, y)
     x_sorted_array = [x, x_position].sort
     y_sorted_array = [y, y_position].sort
 
-    #if piece is in current position
-    if (x == self.x_position) && (y == self.y_position) 
-      false
+    obstructions = game.pieces.find do |chess_piece|
+      next true if has_diagonal_obstruction?(x, y, chess_piece, x_sorted_array, y_sorted_array)
+      next true if chess_piece.x_position.between?(x_sorted_array[0] + 1, x_sorted_array[1] - 1)
 
-    #Vertical
-    elsif y == self.y_position 
-      @pieces_in_the_way = []
-      self.game.pieces.each do |chess_piece|
-        if chess_piece != self
-          if (chess_piece.y_position == y) && chess_piece.x_position.between?(x_sorted_array[0], x_sorted_array[1])
-            @pieces_in_the_way << chess_piece
-          end
-        end
-      end
-      !@pieces_in_the_way.empty?
+      chess_piece.y_position.between?(y_sorted_array[0] + 1, y_sorted_array[1] - 1)
 
-    #Horizontal
-    elsif x == self.x_position 
-      @pieces_in_the_way = []
-      self.game.pieces.each do |chess_piece|    if chess_piece != self
-          if (chess_piece.x_position == x) && chess_piece.y_position.between?(y_sorted_array[0], y_sorted_array[1])
-            @pieces_in_the_way << chess_piece
-          end
-        end
-      end
-      !@pieces_in_the_way.empty?
-
-    #Diagonal
-    elsif (x - self.x_position).abs == (y-self.y_position).abs
-      @pieces_in_the_way = self.game.pieces.select do |chess_piece|
-        next false if chess_piece == self
-        ((x - chess_piece.x_position).abs == (y - chess_piece.y_position).abs) && chess_piece.x_position.between?(x_sorted_array[0], x_sorted_array[1]) && chess_piece.y_position.between?(y_sorted_array[0], y_sorted_array[1])
-      end
-      !@pieces_in_the_way.empty?
+    end
 
+    return obstructions.present? || players_own_piece_is_there?(x, y) 
+    
+  end
+  
+  def has_diagonal_obstruction?(x, y, chess_piece, x_sorted_array, y_sorted_array)
+    is_eq_abs = (x - chess_piece.x_position).abs == (y - chess_piece.y_position).abs
+    is_between_x = chess_piece.x_position.between?(x_sorted_array[0] + 1, x_sorted_array[1] - 1)
+    is_between_y = chess_piece.y_position.between?(y_sorted_array[0] + 1, y_sorted_array[1] - 1)
+    return is_eq_abs && is_between_x && is_between_y
+  end
+  
+  def players_own_piece_is_there?(x, y)
+    occupying_piece = Piece.where(x_position: x, y_position: y, game_id: game.id)
+    if occupying_piece.any? then
+      return occupying_piece.first.is_white? && is_white? || occupying_piece.first.is_black? && is_black?
```

Comment:
So if we only had one method `is_white?` then this could be a ternary statement checking for equality:

`occupying_piece.first.is_white? == is_white?  ? true : false`

Reply:
Even with two methods this would be an improvement.